app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,67 @@
+class MiqCockpitWsWorker::Authenticator
+  def authenticate_for_host(token, host)
```

Comment:
It would be nice to have this method return an object instead of a hash.  Is this really `host_credentials`?  Should this live in the host model as a `credentials_for_token` method?  Note, it looks to assume specific types of authentications, ones without password, such as one with auth_ikey.  I'm not sure how generic this method is, is it valid for other providers such as vmware, google, amazon, etc.?